### DISCUSSION TITLE
Proposal for using folders in the Chart Explorer for metrics and columns

### DISCUSSION THREAD

**Discussion by mattitoo** [apache/superset#19819] [Ideas]
**Motivation**
In the left hand panel of the Chart Explorer all metrics and columns are listed. It is possible to have many columns and metrics, making the flat list hard to use. User can use the search function with good results, but not all users want to do this. Also, discovery of columns or metrics in the dataset is difficult in the current way. Consider this screenshot of 37 metrics, it is already messy and more metrics will come in this use case.
<img width="356" alt="Bildschirmfoto 2022-04-21 um 14 35 01" src="https://user-images.githubusercontent.com/104143980/164717095-6fe4e322-7188-460b-a6cd-f6c3aea1b887.png">

**Proposed Change**
Make it possible for Dataset maintainers to create folders for metrics and columns and assign columns and metrics to these folders. Ideally, an order of the folders and the items within a folder can be determined. 
In the Chart Explorer, these folders are shown as „regular“ folders that can be expanded or collapsed.
-> Additional UX work is needed for this; I would be glad to push this along, but want to see if this topic is interest to the wider Superset user base.

**New or Changed Public Interfaces**
The possibility to add folders should be added to the Dataset 

Editor as a separate tab and/or in the details of a column/metric.
The left hand pane in Chart Explorer should show the folders.
The option to turn this on or off can be placed in the „Feature Flags“ in config.py

**New dependencies**
unknown

**Migration Plan and Compatibility**
unknown

**Alternatives**
Instead of folders, tags or labels could be used. A different way of representation in the front end must be determined for this. A discussion on this might prove helpful, so feel free to post your opinions.

**Comment by mattitoo** [comment]
btw, we can (and probably will) develop this feature, for me it is a matter of seeing if there is interest in the community for this.

**Comment by danisola** [comment]
I also think it's very much needed. Given Superset's [dataset-centric world](https://preset.io/blog/dataset-centric-visualization/), we often create data marts with fact tables and dimensions denormalised in a single dataset. That's great for performance, but it makes it difficult for users to understand and navigate metrics and dimensions.

Here's an example of one of our marts:

<img width="263" alt="image" src="https://user-images.githubusercontent.com/4731758/168600748-4576b494-791c-439d-a229-3b1137e66d17.png">

<img width="267" alt="image" src="https://user-images.githubusercontent.com/4731758/168600804-f3ac63d6-ea8c-42a2-8249-f6d3a8682fdc.png">

IMO, Looker does a good job at allowing some structure:

![image](https://user-images.githubusercontent.com/4731758/168600948-10d79d31-0677-482a-bcea-d7aa9de14395.png)
